src/Documents/Session/IRevisionsSessionOperations.ts

```diff
@@ -0,0 +1,78 @@
+import {IRavenObject} from "../../Types";
+import {DocumentType} from "../DocumentAbstractions";
+import {MetadataAsDictionary} from "../../Mapping/MetadataAsDictionary";
+
+/**
+ * Revisions advanced session operations
+ */
+export interface IRevisionsSessionOperations {
+
+    /**
+     * Returns all previous document revisions for specified document
+     *  ordered by most recent revisions first.
+     * @param id Document id
+     */
+    getFor<TEntity extends object>(id: string): Promise<TEntity[]>;
```

Comment:
I believe we need callback api overloads for all of those

Reply:
done